resip/stack/GenericPidfContents.hxx

```diff
@@ -71,6 +71,7 @@ public:
       EncodeStream& encode(EncodeStream& str, Data indent) const;
    };
    const NodeList& getRootNodes() const { checkParsed(); return mRootNodes; }
+   void setRootNodes(const NodeList& nodeList);
```

Comment:
it would have been better to supply the changes adding the setRoodNodes method as a different commit and a different pull request, that is called making an "atomic commit".  It makes it much easier to review and accept changes incrementally.